FreeRTOS/FreeRTOS-Plus-TCP
# IPV6/Multi : changes to comply to MISRA C 2012. Do not merge

Please do not merge this PR.

Description
-----------
In this PR I present the changes that I made to the IPv6/Multi branch to make it comply with the MISRA C 2012 rules. Now it will comply to all rules of type **mandatory and required**.

The real PR will follow later. This time I want to profit from the many checks like spelling, formatting, etc, see here below.

Test Steps
-----------
Please have a look at the [Coverity project](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/tree/main/test/Coverity) that was recently added by Aniruddha.
I ran it on an Ubuntu machine.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
